crates/telio-utils/src/local_interfaces.rs

```diff
@@ -0,0 +1,91 @@
+use ipnet::Ipv4Net;
+use std::net::{IpAddr, Ipv4Addr};
+
+/// Function type of a function returning interface addresses from the system
+pub type GetIfAddrs = fn() -> std::io::Result<Vec<if_addrs::Interface>>;
+
+/// Function returning interface addresses from the system
+pub fn system_get_if_addr() -> std::io::Result<Vec<if_addrs::Interface>> {
+    if_addrs::get_if_addrs()
+}
+
+/// Method that returns vector of Interfaces on the system
+/// Filtering out meshnet and loopback IP
+pub fn gather_local_interfaces(
+    get_if_addr: GetIfAddrs,
+) -> std::io::Result<Vec<if_addrs::Interface>> {
+    let shared_range: Ipv4Net = Ipv4Net::new(Ipv4Addr::new(100, 64, 0, 0), 10).unwrap_or_default();
+    Ok(get_if_addr()?
+        // .get()?
+        .into_iter()
+        .filter(|x| !x.addr.is_loopback())
+        .filter(|x| match x.addr.ip() {
```

Comment:
What is the benefit of having two separate `filter()`?

Reply:
None 😛